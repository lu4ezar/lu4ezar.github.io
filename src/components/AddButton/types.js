// @flow
import type { Mode, ModeAction } from "../../types/mode";

export type Props = {| setMode: (mode: Mode) => ModeAction |};
