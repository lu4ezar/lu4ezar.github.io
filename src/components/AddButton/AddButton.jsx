// @flow
import * as React from "react";
import AddIcon from "@material-ui/icons/Add";
import Fab from "./styles";
import type { Props } from "./types";

const AddButton = ({ setMode, entity }: Props): React.Node => (
  <Fab color="secondary" title="add todo" onClick={setMode}>
    <AddIcon /> ${entity}
  </Fab>
);

export default AddButton;
