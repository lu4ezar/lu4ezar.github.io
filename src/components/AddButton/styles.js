import { Fab } from "@material-ui/core";
import styled from "styled-components";

const StyledFab = styled(Fab)`
  position: absolute;
  bottom: 1em;
  right: 1em;
`;

export default StyledFab;
