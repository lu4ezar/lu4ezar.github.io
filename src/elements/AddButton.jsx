// @flow
import React from "react";
import { Fab } from "@material-ui/core";
import AddIcon from "@material-ui/icons/Add";

const AddButton = ({ setMode }: { setMode: () => void }) => (
  <Fab
    style={{ position: "absolute", bottom: "1em", right: "1em" }}
    color="secondary"
    onClick={setMode}
  >
    <AddIcon />
  </Fab>
);

export default AddButton;
