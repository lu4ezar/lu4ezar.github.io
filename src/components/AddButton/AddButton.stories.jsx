import React from "react";
import AddButton from "./AddButton";

export default {
  component: AddButton,
  title: "AddButton",
};

const Template = (args) => <AddButton {...args} />;

export const Default = Template.bind({});
Default.args = {};
