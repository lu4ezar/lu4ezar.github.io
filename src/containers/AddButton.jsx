// @flow
import * as React from "react";
import { modeVar } from "../apollo/cache";
import AddButton from "../components/AddButton";

export default () => <AddButton setMode={() => modeVar("form")} />;
