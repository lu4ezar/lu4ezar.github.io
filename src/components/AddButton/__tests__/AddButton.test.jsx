/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { render, fireEvent } from "@testing-library/react";
import AddButton from "../AddButton";

const props = {
  setMode: jest.fn(),
};

describe("Add Button", () => {
  it("renders as expected", () => {
    const { container } = render(<AddButton {...props} />);
    expect(container).toMatchSnapshot();
  });
  it("calls setMode prop function", () => {
    const container = render(<AddButton {...props} />);
    const addButton = container.getByTitle(/add/i);
    fireEvent.click(addButton);
    expect(props.setMode).toBeCalled();
  });
});
