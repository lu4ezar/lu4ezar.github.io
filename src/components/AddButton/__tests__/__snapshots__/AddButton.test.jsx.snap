// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Button renders as expected 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiFab-root sc-bdfBwQ MuiFab-secondary"
    tabindex="0"
    title="add todo"
    type="button"
  >
    <span
      class="MuiFab-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
